iceoryx_binding_c/source/c_wait_set.cpp

```diff
@@ -114,6 +115,7 @@ iox_WaitSetResult iox_ws_attach_subscriber_state(iox_ws_t const self,
                                                  const uint64_t eventId,
                                                  void (*callback)(iox_sub_t))
 {
+    cxx::Expects(callback != nullptr && "Callback function cannot be a nullptr");
```

Comment:
Could you please also check it the C binding for the `Listener` have the same problem?

Reply:
Let's deal with that in a separate PR: https://github.com/eclipse-iceoryx/iceoryx/pull/933